bran/helpers/raven.py

```diff
@@ -0,0 +1,165 @@
+        {
+            'type': 'list',
+            'name': 'instance',
+            'message': 'Select Instance Type',
+            'choices': instance_types
+        },
+        {
+            'type': 'checkbox',
+            'name': 'sg',
+            'message': 'Select all Security Groups',
+            'choices': list_to_choices(sg_names)
+        },
+        {
+            'type': 'input',
+            'name': 'storage',
+            'message': 'Enter storage amount (gb)'
+        },
+        {
+            'type': 'list',
+            'name': 'plugin',
+            'message': 'Select the ravenML plugin you wish to install',
+            'choices': list_to_choices(plugins)
+        },
+        {
+            'type': 'list',
+            'name': 'branch',
+            'message': 'Select the ravenML-train-plugins branch you wish to clone',
+            'choices': list_to_choices(branches)
+        },
+        {
+            'type': 'input',
+            'name': 'config',
+            'message': 'Enter filepath of ravenML config file',
+            'validate': lambda p: os.path.isfile(p)
+        },
+        {
+            'type': 'list',
+            'name': 'gpu',
+            'message': 'Train on GPU or CPU?',
+            'choices': list_to_choices(['GPU', 'CPU'])
+        }
+    ]
+
+    return questions
+
+def get_raven_init_script(plugin, gpu, branch):
+    """
+    Bash script represented as a string that will run on startup in the ec2 
+    instance. Downloads the various requirements for raven and starts a docker 
+    container with AWS credentials injected as environment variables.
+
+    Return:
+        user_data_script (string): The boto3 ec2 create instance method converts
+            the string to a bash script
+    """
+    aws_config = get_local_awsconfig()
+    comet_api_key = get_comet_api_key()
+   
+    ## the following commands are potentially useful but are not needed/do not work atm. saving for future use
+    # pip install "git+https://github.com/autognc/ravenML-train-plugins.git#egg=rmltraintfbbox&subdirectory=rmltraintfbbox" - pip install from subdirectory of github repo
+    # source /home/ubuntu/anaconda3/bin/activate /home/ubuntu/anaconda3/envs/ravenml - enter conda env that belongs to a different user
+    
+    # this script runs config for ravenML upon startup some things to note:
+    # ubuntu deep learning AMIs run the user data script as the root user, however aws makes you connect as ubuntu
+    # therefore, one should use chown -R ubuntu:ubuntu <filepath> on any file/directory that is created in this script 
+    # aws docs say not to use sudo as a prefix to any command in this script
+    user_data_script = """#!/bin/bash
+    # direct stdout to /var/log/user-data.log
+    exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1
+    
+    # set environment variables
+    echo "export EC2_ID=$(echo $(curl http://169.254.169.254/latest/meta-data/instance-id))" >> /etc/profile
```

Comment:
Where is this IP coming from?

Reply:
hmm I actually have no clue. I'll look into this

Reply:
That's a generic EC2 metadata IP that works on any EC2 instance. Exporting the environment variable isn't actually necessary anymore, because we switched over to using the IP directly to grab the ID when shutting down the instance.